utils/merkletrie/index/node.go

```diff
@@ -0,0 +1,113 @@
+package index
```

Comment:
this should probably no go under the merkletrie package, but on the index.

Please document extensively how are you planning to map an index to a merkletrie, the challenges, surprises and your solutions. Otherwise, I can only guess your intentions and it is very time consuming and error prone.